zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -355,6 +428,102 @@ impl DiskWriteBatch {
             self.zs_insert(&orchard_tree_cf, height, trees.orchard);
         }
 
+        if let Some(subtree) = trees.sapling_subtree {
+            let (subtree_end, subtree_start) = subtree.bounds();
+
+            let start_height = if subtree_end.index == 0 {
+                // the first entry after Height(0) should be the height of the first Sapling note
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(Height(1)..)
+                    .next()
+                    .expect("must have sapling notes on disk");
+
+                start_height
+            } else if let Some(start_height) =
+                zebra_db.sapling_subtree_start_height_by_index(subtree_end.index)
```

Comment:
> How can we make sure the upgrade runs before this code runs?

We don't have to anymore since we're not tracking start boundaries, if we want to add those later, it could find the start height from the note commitment trees when there's no subtree at the previous index.